### PR TITLE
Implement legacy group and role separation behavior into the roles section in user settings page

### PR DESCRIPTION
**Pull request by nipunsampath**
### Purpose
Fixes https://github.com/wso2/product-is/issues/18002 by extracting roles from the groups section in SCIM user response when the legacy group and role separation behavior is enabled.

### Related Issues
- https://github.com/wso2/product-is/issues/18002

### Checklist
- [ ] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/users/components/user-roles-list.tsx [hunk 1]
```diff
@@ -54,17 +62,57 @@ export const UserRolesList: FunctionComponent<UserRoleEditPropsInterface> = (
...
-            setInitialSelectedRolesOptions(user.roles);
+        let userRoles: RolesMemberInterface[];
+        
+        if (isGroupAndRoleSeparationEnabled && user?.roles?.length > 0) {
+            userRoles = user.roles;
+        } else {
+            userRoles = extractUserRolesFromGroups(user?.groups);
+        }
+
+        if ( userRoles?.length > 0 ) {
```
Comment (brionmario) [inline apps/console/src/features/users/components/user-roles-list.tsx:80]:
```suggestion
        if (userRoles?.length > 0) {
```

File: apps/console/src/features/users/components/user-roles-list.tsx [hunk 2]
```diff
@@ -54,17 +62,57 @@ export const UserRolesList: FunctionComponent<UserRoleEditPropsInterface> = (
...
+            setInitialSelectedRolesOptions(userRoles);
         } else {
             setShowEmptyRolesListPlaceholder(true);
         }
+
```
Comment (pavinduLakshan) [inline apps/console/src/features/users/components/user-roles-list.tsx:85]:
```suggestion
```

File: apps/console/src/features/users/components/user-roles-list.tsx [hunk 3]
```diff
@@ -54,17 +62,57 @@ export const UserRolesList: FunctionComponent<UserRoleEditPropsInterface> = (
...
+     * 
+     * @param groups - Groups of the user
+     * @returns Roles of the user
+     */
+    const extractUserRolesFromGroups = (groups: RoleGroupsInterface[]): RolesMemberInterface[] => {
+
+        const userRoles: RolesMemberInterface[] = [];
+        
+        groups?.forEach((group: RoleGroupsInterface) => {
+            const displayName: string[] = group?.display?.split(DOMAIN_SEPARATOR);
```
Comment (pavinduLakshan) [inline apps/console/src/features/users/components/user-roles-list.tsx:100]: The variable name displayName doesn't convey that its an array. Shall we use a more comprehensive variable name?

File: apps/console/src/features/users/components/user-roles-list.tsx (hunk 1 above)
Comment (nipunsampath) [reply apps/console/src/features/users/components/user-roles-list.tsx]: Fixed with 3826c3d46538353a2305821cf8b0e970359c9c7b

File: apps/console/src/features/users/components/user-roles-list.tsx (hunk 2 above)
Comment (nipunsampath) [reply apps/console/src/features/users/components/user-roles-list.tsx]: Fixed with 3826c3d46538353a2305821cf8b0e970359c9c7b

File: apps/console/src/features/users/components/user-roles-list.tsx [hunk 4]
```diff
@@ -54,17 +62,56 @@ export const UserRolesList: FunctionComponent<UserRoleEditPropsInterface> = (
...
+     * 
+     * @param groups - Groups of the user
+     * @returns Roles of the user
+     */
+    const extractUserRolesFromGroups = (groups: RoleGroupsInterface[]): RolesMemberInterface[] => {
+
+        const userRoles: RolesMemberInterface[] = [];
+        
+        groups?.forEach((group: RoleGroupsInterface) => {
+            const splitDisplayName: string[] = group?.display?.split(DOMAIN_SEPARATOR);
```
Comment (pavinduLakshan) [inline apps/console/src/features/users/components/user-roles-list.tsx:99]: We can improve readability further as follows.

```suggestion
            const [displayName, userStoreDomain ] = group?.display?.split(DOMAIN_SEPARATOR);
```

just a suggestion :)

**Review by pavinduLakshan — DISMISSED** [review]

File: apps/console/src/features/users/components/user-roles-list.tsx (hunk 4 above)
Comment (nipunsampath) [reply apps/console/src/features/users/components/user-roles-list.tsx]: Resolved with 3741ac80f563720a212d984785f0001795df74d8

**Review by pavinduLakshan — APPROVED** [review]

**Pull request merged**